File: chrome/test/data/webui/bookmarks/bookmarks_focus_test.js
// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Tests for Bookmarks which are run as interactive ui tests.
 * Should be used for tests which care about focus.
 */

GEN_INCLUDE(['//chrome/test/data/webui/polymer_interactive_ui_test.js']);

GEN('#include "content/public/test/browser_test.h"');

const BookmarksFocusTest = class extends PolymerInteractiveUITest {
  /** @override */
  get browsePreload() {
    throw new Error('this is abstract and should be overriden by subclasses');
  }
};

var BookmarksFolderNodeFocusTest = class extends BookmarksFocusTest {
  /** @override */
  get browsePreload() {
    return 'chrome://bookmarks/test_loader.html?module=bookmarks/folder_node_focus_test.js';
  }
};

// http://crbug.com/1000950 : Flaky.
GEN('#define MAYBE_All DISABLED_All');
TEST_F('BookmarksFolderNodeFocusTest', 'MAYBE_All', function() {
  mocha.run();
});

var BookmarksListFocusTest = class extends BookmarksFocusTest {
  /** @override */
  get browsePreload() {
    return 'chrome://bookmarks/test_loader.html?module=bookmarks/list_focus_test.js';
  }
};

// http://crbug.com/1000950 : Flaky.
GEN('#define MAYBE_All DISABLED_All');
TEST_F('BookmarksListFocusTest', 'MAYBE_All', function() {
  mocha.run();
});

var BookmarksDialogFocusManagerTest = class extends BookmarksFocusTest {
  /** @override */
  get browsePreload() {
    return 'chrome://bookmarks/test_loader.html?module=bookmarks/dialog_focus_manager_test.js';
  }
};

// http://crbug.com/1000950 : Flaky.
GEN('#define MAYBE_All DISABLED_All');
TEST_F('BookmarksDialogFocusManagerTest', 'MAYBE_All', function() {
  mocha.run();
});

File: chrome/test/data/webui/bookmarks/bookmarks_focus_test.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

class BookmarksFocusTest : public WebUIMochaFocusTest {
 protected:
  BookmarksFocusTest() { set_test_loader_host(chrome::kChromeUIBookmarksHost); }
};

IN_PROC_BROWSER_TEST_F(BookmarksFocusTest, FolderNode) {
  RunTest("bookmarks/folder_node_focus_test.js", "mocha.run()");
}

// http://crbug.com/1000950 : Flaky.
IN_PROC_BROWSER_TEST_F(BookmarksFocusTest, DISABLED_List) {
  RunTest("bookmarks/list_focus_test.js", "mocha.run()");
}

// http://crbug.com/1000950 : Flaky.
IN_PROC_BROWSER_TEST_F(BookmarksFocusTest, DISABLED_DialogFocusManager) {
  RunTest("bookmarks/dialog_focus_manager_test.js", "mocha.run()");
}

// TODO(crbug.com/40889088): Test is flaky.
IN_PROC_BROWSER_TEST_F(BookmarksFocusTest, DNDManager) {
  RunTest("bookmarks/dnd_manager_test.js", "mocha.run()");
}
